src/operations/aggregate.ts

```diff
@@ -47,14 +42,11 @@ export class AggregateOperation<T = Document> extends CommandOperation<T> {
   pipeline: Document[];
   hasWriteStage: boolean;
 
-  constructor(parent: OperationParent, pipeline: Document[], options?: AggregateOptions) {
-    super(parent, options);
+  constructor(pipeline: Document[], ns?: MongoDBNamespace, options?: AggregateOptions) {
```

Comment:
Don't change this if it turns out to be a lot of refactor but: AggOp takes pipeline then namespace, and AggCursor takes namespace then pipeline. Seems odd that those would be different unless one is consistent with the cursor and the other is consistent with operations. Any insight there and if its worth a little refactor?

Reply:
I can go ahead and fix that actually just for dev friendliness. Shouldn't take too long. I did it that way to have the namespace be optional for the AggregateOperation, but since we're axing the parent parameter, it should just be mandatory and be the first argument.